thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/monitor/MonitorJobRunner.java

```diff
@@ -0,0 +1,160 @@
+package com.linkedin.thirdeye.anomaly.monitor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.context.internal.ManagedSessionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.linkedin.thirdeye.anomaly.job.JobConstants.JobStatus;
+import com.linkedin.thirdeye.anomaly.job.JobContext;
+import com.linkedin.thirdeye.anomaly.task.TaskConstants.TaskStatus;
+import com.linkedin.thirdeye.anomaly.task.TaskConstants.TaskType;
+import com.linkedin.thirdeye.anomaly.task.TaskGenerator;
+import com.linkedin.thirdeye.detector.db.dao.AnomalyJobSpecDAO;
+import com.linkedin.thirdeye.detector.db.dao.AnomalyTaskSpecDAO;
+import com.linkedin.thirdeye.detector.db.entity.AnomalyJobSpec;
+import com.linkedin.thirdeye.detector.db.entity.AnomalyTaskSpec;
+
+public class MonitorJobRunner implements Runnable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(MonitorJobRunner.class);
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  private AnomalyJobSpecDAO anomalyJobSpecDAO;
+  private AnomalyTaskSpecDAO anomalyTaskSpecDAO;
+  private SessionFactory sessionFactory;
+  private MonitorConfiguration monitorConfiguration;
+  private TaskGenerator taskGenerator;
+  private JobContext jobContext;
+
+  public MonitorJobRunner(AnomalyJobSpecDAO anomalyJobSpecDAO, AnomalyTaskSpecDAO anomalyTaskSpecDAO,
+      SessionFactory sessionFactory, MonitorConfiguration monitorConfiguration) {
+    this.anomalyJobSpecDAO = anomalyJobSpecDAO;
+    this.anomalyTaskSpecDAO = anomalyTaskSpecDAO;
+    this.sessionFactory = sessionFactory;
+    this.monitorConfiguration = monitorConfiguration;
+
+    taskGenerator = new TaskGenerator();
+    jobContext = new JobContext();
+  }
+
+  @Override
+  public void run() {
+    try {
+      LOG.info("Starting monitor job");
+
+      List<AnomalyJobSpec> anomalyJobSpecs = findAnomalyJobsWithStatusScheduled();
+      jobContext.setJobName(TaskType.MONITOR.toString());
+      long jobExecutionId = createAnomalyJob();
+      jobContext.setJobExecutionId(jobExecutionId);
+      List<Long> taskIds = createAnomalyTasks(anomalyJobSpecs);
+
+    } catch (Exception e) {
+      LOG.error("Exception in monitor job runner", e);
+    }
+  }
+
+  private long createAnomalyJob() {
+    Session session = sessionFactory.openSession();
+    Long jobExecutionId = null;
+    try {
+      ManagedSessionContext.bind(session);
+      Transaction transaction = session.beginTransaction();
+      try {
+        LOG.info("Creating monitor job");
+        AnomalyJobSpec anomalyJobSpec = new AnomalyJobSpec();
```

Comment:
Trying having a parent class called JobSpec. AnomalyJobSpec and MonitorJobSpec should inherit from the parent class